### ISSUE TITLE
Mac: AppleSpell process never finishes, number of errors bounces back and forth.

### ISSUE THREAD

**Issue by GandalfSaxe** [bartosz-antosik/vscode-spellright#195]
Hi,

As the title says, the errors/files found keeps bouncing back and fourth and never finishes (so AppleSpell keep running in the background at 40-60% CPU), when I'm in the Problems panel. If I go to Output, Debug or Terminal, it stops.

I don't have to time right now to narrow this issue down to a minimum example. Is there a debug log I can read something off somewhere you give us a better idea of the case? Otherwise I can simply provide you with my workspace. It's a 20 MB LaTeX project and contains nothing secret.

**Comment by bartosz-antosik** [comment]
You could show what is happening in Developer Tools (Help menu) console. Enabling logging will probably not show anything important but you can do this by setting SPELLRIGHT_DEBUG_OUTPUT in src/extension.js. But I do not expect too much information from this log.

If you could maybe do the following test it may be a lot of help if what happens is a result of an interaction with some other extension: if you could install [portable edition](https://code.visualstudio.com/docs/editor/portable) and then install just Spell Right and see if the problem still happens.

**Comment by bartosz-antosik** [comment]
Oh, and if you could share the project would be lovely - I could see if I can reproduce it on other OSes (I do not have macOS).

**Comment by GandalfSaxe** [comment]
I just tried to recreate it again. I had to disable SpellRight, reload VSC, enable it, and build the project a couple of times, not sure why but it happened again as shown here in the GIF.

Note:
- A file keeps popping in and out of the Problems view
- The Error keeps bouncing back and fourth between 0 and 1 in the lower left corner
- AppleSpell process consistently sit at 60% CPU

![](https://i.imgur.com/WyF3TpK.gif)

Here is the report: https://transfer.sh/KKdlW/report.zip

Open Thesis.tex and try to build it, make a small change and save etc. If it doesn't appear, re-enabling SpellRight.

**Comment by bartosz-antosik** [comment]
I see, but have you maybe tried to recreate this just with Spell Right enabled?

**Comment by GandalfSaxe** [comment]
I'll do that now.

PS: It does happen consistently, I just need to build the project before Spell Right is activated.

**Comment by GandalfSaxe** [comment]
Yes, also happens with just LaTeX Workshop and Spell Right installed (I can't built without LaTeX Workshop so I don't know how I would activate Spell Right / check without that)

![](https://i.imgur.com/VB2P4Rp.gif)

**Comment by bartosz-antosik** [comment]
Thank you for the test.

**Comment by bartosz-antosik** [comment]
I really have no clue. Could you please show me your `.vscode` file and user lever configuration of Spell Right and Latex Workshop?

**Comment by GandalfSaxe** [comment]
Workspace settings:

```json
{
    "cSpell.enabled": true,
    "spellright.language": "en",
    "spellright.documentTypes": [
        "markdown",
        "plaintext",
        "latex"
    ],
    "cSpell.words": [
        "Adair",
        "Analytique",
        "Belbruno",
        "Dimensionalization",
        "Elon",
        "Hiten",
        "Hohmann",
        "Hohmann's",
        "Keplerian",
        "Kutta",
        "LETO",
        "LETOs",
        "Mathematica",
        "Mécanique",
        "Noether's",
        "PDEs",
        "Raphson",
        "Ratliffe",
        "Runge",
        "Störmer",
        "Topputo",
        "Tuppoto",
        "Verlet",
        "apoapsis",
        "apocenter",
        "dimensionalize",
        "dimensionalized",
        "discretize",
        "eigenmode",
        "equipotential",
        "integrators",
        "parallelization",
        "periapsis",
        "pericenter",
        "prograde",
        "roadmap",
        "symplectic",
        "xelatex",
        "xlabel",
        "ylabel",
        "yscale"
    ]
}
```

User settings:

```json
{
  "latex-workshop.latex.autoBuild.onSave.enabled": true,
  "workbench.iconTheme": "vscode-icons",
  "vsicons.dontShowNewVersionMessage": true,
  "git.autofetch": true,
  "explorer.sortOrder": "type",
  "window.restoreWindows": "all",
  "[json]": {
    "editor.detectIndentation": true,
    "editor.tabSize": 2
  },
  "[markdown]": {
    "editor.detectIndentation": true,
    "editor.tabSize": 2
  },
  // markdownlint config object
  "markdownlint.config": {
    "MD033": false, // no-inline-html
    "MD026": {
      "punctuation": ".,;:!"
    }, // no-trailing-punctuation (allow "?" in headers)
    "MD040": false // fenced-code-language
  },
  "[python]": {
    "editor.formatOnSave": false,
    "editor.codeActionsOnSave": {
      "source.organizeImports": false
    },
    "editor.rulers": [
      88
    ]
  },
  "python.formatting.provider": "black",
  "[latex]": {
    "editor.wordWrap": "on"
  },
  "markdown-preview-enhanced.pandocMarkdownFlavor": "markdown",
  "markdown-preview-enhanced.pandocArguments": "--filter=pandoc-crossref, --filter=pandoc-citeproc, --filter=pandoc-latex-newpage",
  // Enable this option will render markdown by pandoc instead of remarkable.
  "markdown-preview-enhanced.usePandocParser": true,
  "pandoc.docxOptString": "",
  "pandoc.htmlOptString": "",
  "pandoc.pdfOptString": "-F pandoc-crossref -F pandoc-citeproc -F pandoc-latex-newpage",
  "terminal.integrated.shell.windows": "C:\\WINDOWS\\System32\\WindowsPowerShell\\v1.0\\powershell.exe",
  // Ignore pandoc-citeproc citations
  "cSpell.languageSettings": [
    {
      "languageId": "markdown",
      "ignoreRegExpList": [
        "\\[@.+\\]/g"
      ]
    }
  ],
  // // Ignore content of code fences in markdown
  // "cSpell.languageSettings": [
  //   {
  //     "languageId": "markdown",
  //     "ignoreRegExpList": [
  //       "/^```(?:.|\\s)+?^```/mig"
  //     ]
  //   }
  // ],
  "slack.teamToken": "CENSORED",
  "slack.username": "CENSORED",
  "slack.avatarUrl": "CENSORED",
  "workbench.colorTheme": "Monokai",
  "latex-workshop.view.pdf.viewer": "tab",
  "diffEditor.renderSideBySide": false,
  "python.jediEnabled": false,
  //MAC ONLY
  "editor.fontFamily": "Fira Code",
  "editor.fontLigatures": true,
  "gitlens.advanced.messages": {
    "suppressShowKeyBindingsNotice": true
  },
}
```

**Comment by GandalfSaxe** [comment]
Now reproduced on Windows too:

https://imgur.com/5Ie9sh8

**Comment by bartosz-antosik** [comment]
Exactly the same way?

**Comment by GandalfSaxe** [comment]
That was with all extensions. Now also reproduced with only LaTeX workshop + Spell Right:

https://gfycat.com/unpleasantuniqueboto

Btw I don't see any runaway spell check processes as AppleSpell @ 60% CPU on Mac, just a Runtime Broker which I suspect is doing the spell checking (not sure) at around 4-5%. But the problem with the detected problems flickering back and forth is still there. Also if I disable LaTeX Workshop and just go into some of the .tex files individually, it detects the problems but it doesn't do the looping.

So it happens on both platforms and seems to be related to LaTeX Workshop building.

**Comment by bartosz-antosik** [comment]
Somehow I cannot zoom these pictures to 100% but do I see correctly that you have only projects main file opened in a tab and what is flickering are other documents which seem to be linted by Latex Workshop?

**Comment by bartosz-antosik** [comment]
I am terribly sorry but could you please make some prescription what to do step-by-step to reproduce this on Windows? I am playing with these two nut somehow I cannot reproduce this behavior.

**Comment by GandalfSaxe** [comment]
What I did was just:

1. Open report folder in VSCode
2. Open Thesis.tex   (but I think any of the involved .tex files would do)
3. Build project with LaTeX Workshop (shortcut: Ctrl+Alt+B)
4. Bug starts manifesting once the PDF has been built.

**Comment by bartosz-antosik** [comment]
Thanks. Sadly, I still cannot reproduce the problem.

**Comment by GandalfSaxe** [comment]
Hm, strange.

**Comment by bartosz-antosik** [comment]
I have attempted few times to recreate the situation in Code portable edition with only Spell Right & Latex Workshop extensions installed: In the first posts you said something about disabling & re-enabling Spell Right: do you mean in extensions panel or just with the switch in status bar?

**Comment by GandalfSaxe** [comment]
> Do you have anything in the .vscode folder in report folder? I have not get anything inside the ZIP.
I have a `spellright.dict` and a `settings.json` with:
```json
{
    "cSpell.enabled": true,
    "spellright.language": "en",
    "spellright.documentTypes": [
        "markdown",
        "plaintext",
        "latex"
    ],
    "cSpell.words": [
    ... (long list of words here)
    ]
}
```
But since Code Spell is turned off in the tests, these settings shouldn't matter.

I meant disabling/enabling the extensions themselves. But I don't know if it's necessary, since then I found it to happen every time I build. Are you sure that the PDF is actually building al the way? I.e. you could try to remove Thesis.pdf and see if it gets created again.

**Comment by bartosz-antosik** [comment]
Yes, `Thesis.pdf` does build. If e.g. I delete it. But it does not start this oscillations (generating spelling diagnostics/switching documents etc.) that your animations show.

**Comment by GandalfSaxe** [comment]
I’m using MikTex on windows (portable). It shouldn’t be the latex distribution but I’m running out of ideas.

**Comment by bartosz-antosik** [comment]
Funny i was just going to ask about this! I am using regular MikTeX.

I do not have any further ideas as for now.

You could maybe debug this a bit? Or maybe set SPELLRIGHT_DEBUG_OUTPUT mentioned in second post to true and see what shows in Console of Developer Tools.

**Comment by bartosz-antosik** [comment]
Don't you think it could be this: [56317](https://github.com/Microsoft/vscode/issues/56317)?

**Comment by bartosz-antosik** [comment]
Hi again! It seems that the above mentioned issue [56317](https://github.com/Microsoft/vscode/issues/56317) has been addressed in version 1.26.1. Could you tell me if you see any change?

**Comment by GandalfSaxe** [comment]
Sorry, I got thesis busy.

I'm on the latest 1.27 Insiders, and I cannot reproduce the problem anymore, so it does seem to have been fixed 🙂 🎉
